### PR TITLE
feat: Lit protocol Mint capacity credit NFT [DEV-4280]

### PR DESCRIPTION
**Pull request by DaevMithran**

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-4280 PoA for using Lit token for each call to Lit Network](https://app.clickup.com/t/6600954/DEV-4280)

File: src/dkg-threshold/lit-protocol/v6.ts [hunk 1]
```diff
@@ -131,16 +148,41 @@ export class LitProtocol {
 		stringHash: string,
 		unifiedAccessControlConditions: NonNullable<UnifiedAccessControlConditions>
 	): Promise<string> {
-		// generate auth signature
-		const authSig = await LitProtocol.generateAuthSignature(this.cosmosAuthWallet);
+		// mint capacity credits
+		const { capacityTokenIdStr } = await this.contractClient.mintCapacityCreditsNFT({
```
Comment (DaevMithran) [inline src/dkg-threshold/lit-protocol/v6.ts:152]: @Eengineer1 can I separate minting capacity credits in a separate function and accept only `sessionSigs` / `capacityDelegationAuthSig` as a parameter to allow token to be minted by a separate entity and delegated to the current wallet address

Comment (Eengineer1) [reply src/dkg-threshold/lit-protocol/v6.ts:154]: Please, synchronise your work with the `capacity-mint` branch, which has all the changes necessary to instantiate the underlying wallet.

We'll review the end result, but before proceeding please, prioritise the modifications present within that branch.

Comment (Eengineer1) [reply src/dkg-threshold/lit-protocol/v6.ts:154]: The question is valid, in which case, already accounted for in the aforementioned implementation branch.

File: src/dkg-threshold/lit-protocol/v6.ts
Comment (Eengineer1) [inline src/dkg-threshold/lit-protocol/v6.ts:1]: Please, follow `capacity-mint` branch's logic, which has decoupled most of Lit's BLS network client from hard assumptions present here.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
Let's follow the relevant listed branch + bring the logic across here.

Otherwise, this PR is less than a quarter of the work necessary to decouple certain internally instantiated objects, such as Lit BLS network client, access to wallet-specific functionality.

Let's reiterate on it when the aforementioned synchronisation has been successfully performed.

File: src/agent/ICheqd.ts [hunk 2]
```diff
@@ -6739,6 +6826,11 @@ export class Cheqd implements IAgentPlugin {
 			);
 		}
 
+		// validate dkgOptions
+		if (!options?.topArgs?.dkgOptions || !options.topArgs?.dkgOptions?.capacityDelegationAuthSignature) {
```
Comment (DaevMithran) [inline src/agent/ICheqd.ts:6830]: @Eengineer1 should I rename `capacityDelegationAuthSignature` to something simpler

Comment (Eengineer1) [reply src/agent/ICheqd.ts:6830]: No need as it captures the necessary semantics, no extra information.

**Review by Eengineer1 — APPROVED** [review]
The core work looks good.

Approved.

**Pull request merged**